x-pack/plugins/lens/common/expressions/time_scale/time_scale.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment-timezone';
+import { i18n } from '@kbn/i18n';
+import type {
+  ExpressionFunctionDefinition,
+  Datatable,
+} from '../../../../../../src/plugins/expressions/common';
+import {
+  getDateHistogramMetaDataByDatatableColumn,
+  parseInterval,
+  calculateBounds,
+} from '../../../../../../src/plugins/data/common';
+import { buildResultColumns } from '../../../../../../src/plugins/expressions/common';
+import type { TimeScaleUnit } from './types';
+
+export interface TimeScaleArgs {
+  dateColumnId: string;
+  inputColumnId: string;
+  outputColumnId: string;
+  targetUnit: TimeScaleUnit;
+  outputColumnName?: string;
+}
+
+const unitInMs: Record<TimeScaleUnit, number> = {
+  s: 1000,
+  m: 1000 * 60,
+  h: 1000 * 60 * 60,
+  d: 1000 * 60 * 60 * 24,
+};
+
+export const timeScale: ExpressionFunctionDefinition<
```

Comment:
So you're no longer passing in the `dataPluginPublicStart` object which was previously used to get the auto interval, you're directly importing it?

Reply:
Correct. I had a look at the code and assumed there were no strict dependencies in the used function with its context, and that the function could have been used independently.